### PR TITLE
Config overhaul, fix crash

### PR DESCRIPTION
**Pull request by BoogieMonster1O1**
- Support for modmenu
- Dedicated server crash fix
- General code verbosity reductions
- Replace comments with tooltips
- Support for accessing the config screen without modmenu

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Idea
+.idea/
+
+# Garbage
+.DS_Store
```
Comment (YTG1234) [inline .gitignore:5]: Was very much needed in my opinion

File: build.gradle [hunk 2]
```diff
@@ -21,15 +21,15 @@ repositories{
 }
 
 dependencies {
-	minecraft "com.mojang:minecraft:${project.minecraft_version}"
-	mappings "net.fabricmc:yarn:${project.yarn_mappings}:v2"
-	modImplementation "net.fabricmc:fabric-loader:${project.loader_version}"
+	minecraft("com.mojang:minecraft:${project.minecraft_version}")
+	mappings("net.fabricmc:yarn:${project.yarn_mappings}:v2")
+	modImplementation("net.fabricmc:fabric-loader:${project.loader_version}")
```
Comment (YTG1234) [inline build.gradle:23]: No why

**Review by YTG1234 — APPROVED** [review]

File: build.gradle (hunk 2 above)
Comment (BoogieMonster1O1) [reply build.gradle:23]: why not

File: .gitignore (hunk 1 above)
Comment (BoogieMonster1O1) [reply .gitignore:5]: obviously 😆

**Comment by redcreeper14385** [comment]
It's early morning at the moment for me, I will review the changes in about an hour.

File: build.gradle (hunk 2 above)
Comment (YTG1234) [reply build.gradle:23]: the buildscript is not Kotlin, you don't have to do it like Kotlin. It looks a little cluttered in my opinion

File: build.gradle [hunk 3]
```diff
@@ -41,10 +38,12 @@ dependencies {
 	modApi("me.shedaniel.cloth:config-2:4.7.0-unstable") {
         exclude(group: "net.fabricmc.fabric-api")
     }
-	include "me.shedaniel.cloth:config-2:4.7.0-unstable"
+	include("me.shedaniel.cloth:config-2:4.7.0-unstable")
```
Comment (YTG1234) [inline build.gradle:41]: no

Comment (BoogieMonster1O1) [reply build.gradle:41]: yes

File: build.gradle (hunk 2 above)
Comment (redcreeper14385) [reply build.gradle:23]: Why the brackets?

File: src/main/java/mounderfod/moundertweaks/MounderTweaksClient.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
+package mounderfod.moundertweaks;
+
```
Comment (redcreeper14385) [inline src/main/java/mounderfod/moundertweaks/MounderTweaksClient.java:2]: Needs to be 1.2.0, 1.1.0 was already released.

**Review by redcreeper14385 — CHANGES_REQUESTED** [review]
Change the version to 1.2.0 please.

File: gradle.properties [hunk 5]
```diff
@@ -8,10 +8,11 @@ org.gradle.jvmargs=-Xmx1G
 	loader_version=0.9.2+build.206
 
 # Mod Properties
-	mod_version = 1.0.0
+	mod_version = 1.1.0
```
Comment (redcreeper14385) [inline gradle.properties]: Needs to be 1.2.0

**Review by redcreeper14385 — CHANGES_REQUESTED** [review]
Reissuing changes request, I set it on the wrong line.

File: gradle.properties (hunk 5 above)
Comment (BoogieMonster1O1) [reply gradle.properties]: :done:

**Comment by redcreeper14385** [comment]
gradle.properties is fixed now, thanks

**Comment by BoogieMonster1O1** [comment]
*why the heck is there crlf everywhere*
*please use lf*

**Comment by BoogieMonster1O1** [comment]
`^M` = concern

**Comment by YTG1234** [comment]
set `.gitattributes` to only to LF and do CRLF only in `gradlew.bat`

File: src/main/java/mounderfod/moundertweaks/MounderTweaksConfig.java [hunk 6]
```diff
@@ -1,36 +0,0 @@
-package mounderfod.moundertweaks;
```
Comment (redcreeper14385) [inline src/main/java/mounderfod/moundertweaks/MounderTweaksConfig.java:1]: @BoogieMonster1O1 why was this file deleted?

**Comment by BoogieMonster1O1** [comment]
> set `.gitattributes` to only to LF and do CRLF only in `gradlew.bat`

yup.
will do in the next pee arr

**Review by redcreeper14385 — APPROVED** [review]
Looks good, will do a final check before merging into master.

**Comment by YTG1234** [comment]
DON'T SAY THE ARR WORD

File: src/main/java/mounderfod/moundertweaks/MounderTweaksConfig.java (hunk 6 above)
Comment (BoogieMonster1O1) [reply src/main/java/mounderfod/moundertweaks/MounderTweaksConfig.java:1]: it was not. it was only moved

**Review by YTG1234 — APPROVED** [review]
It's good.

**Comment by redcreeper14385** [comment]
May I ask why you redid the config file entirely? Also why did you replace SandBlockMixin with FallingBlockMixin?

**Comment by BoogieMonster1O1** [comment]
> May I ask why you redid the config file entirely? Also why did you replace SandBlockMixin with FallingBlockMixin?

added tooltips for the config. and redid the entire mixin. everything works.

**Comment by redcreeper14385** [comment]
Ok we should be ready to merge, any objections before I do?

**Comment by BoogieMonster1O1** [comment]
> Ok we should be ready to merge, any objections before I do?

no
